src/sema/diagnostics.rs

```diff
@@ -167,8 +167,6 @@ fn convert_diagnostic(
                 diagnostic::Label::secondary(file_id[&file_no], start..end)
                     .with_message(note.message.to_owned()),
             );
-        } else {
-            unreachable!("note without file position");
```

Comment:
Why did you remove this?

Reply:
We can't panic just because a note doesn't point to a file location. Builtins don't have a file location, so we just ignore the note

Reply:
If a note has a builtin file location the note should be rendered differently. I don't think we generate them.

Silently dropping them does not seem like a good idea.

Reply:
The `diagnostics::Label` structs needs a file location. Feels like this would require some refactoring best done in a dedicated PR?

I don't think this is a huge problem, as it only occurs for our builtins and it only drops the note (the primary diagnostic is still there).

I can revert this change if you want. It's not specific to this chain extension builtin. I discovered it by accident and thought, instead of crashing the compiler, it'd be nicer to just not display the secondary note.

Reply:
I don't agree that silently dropping something is better than a panic. If there is a problem, we would like to know about it. The earlier we catch a problem the better.

Reply:
It makes for a bad user experience. Given this code:

```solidity
import "substrate";
contract Foo {
   function chain_extension() public pure {}
}
```

Would you rather have 

```
thread 'main' panicked at 'internal error: entered unreachable code: note without file position', src/sema/diagnostics.rs:171:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Instead of this?

```
warning: chain_extension is already defined as a function
  ┌─ /home/glow/code/solang/test.sol:3:13
  │
3 │    function chain_extension() public pure {}
  │             ^^^^^^^^^^^^^^^
```

I mean we are already aware of the problem. I'll open an issue so we don't loose track.

Reply:
> The `diagnostics::Label` structs needs a file location. Feels like this would require some refactoring best done in a dedicated PR?

A dedicated PR would be the best option, so it is better now to revert this change if it does not affect this PR in any way.

Reply:
See #1308

Reply:
I don't see how it is better to panic. This is note about the diagnostic itself. We are not silently swallowing an error. We are just omitting a note containing additional information.

Reply:
Removing the panic is valid fix.

A panic is a last resort; you have an error that you can't do anything about it. Hence you decide to crash your program.

This is just wrong here. If we can't print the note because it makes no sense to print it. Then we just should not print it.

But anyways, it's not important for this PR. I reverted this change.